whitepaper/chapter01.md

```diff
@@ -0,0 +1,150 @@
+
+<a name="1.3.3"></a>
+### 1.3.3 Infrastructure, Deployment and Configuration of Cloud Native Systems
+
+[Immutable infrastructure](https://docs.google.com/document/d/1fwV3vZB5IgHb6uUgQpHts9mFR0HheYbIrlg9hGfWYfA) (the orchestrator and all of the software and hardware that it depends on) is provisioned using baked and versioned templates (e.g. server images) or a combination of templates and bootstrapping (some repeatable and versioned process that is applied to the template e.g. Kubeadm).  The underlying infrastructure is not changed after it is made ready for use. New changes to the infrastructure are rolled out as new instances of infrastructure. By separating the application layer from the infrastructure layer, it is possible to iterate faster with minimal dependency risk. Similarly it enables the vision of programmable and software defined infrastructure that can made possible to automate end to end deployment .
+
+In cloud native systems, the atomic unit of deployment is the container, on an immutable and decoupled compute environment that nearly completely abstracts function from the underlying physical infrastructure and network architecture. At present, the most commonly used container environment is an OCI-compliant implementation. A container is different than a Virtual Machine (VM), which was the initial base unit of cloud computing. A VM hosts a Guest OS capable of delivering a service's required compute and networking. This guest runs on top of a hypervisor consuming the virtual infrastructure and orchestration it provides. In contrast, containers run on top of a shared Host OS (with a shared kernel) kernel. This distinction is key because containerisation enables immutability, disposability and resource efficiency in ways that are not easily achievable with VMs.
+
+For Kubernetes specifically, the atomic unit of management is a Pod. A Pod is a set of containers co-located in the same or different host and sharing the same Linux networking namespace.  For telco (and other) cloud native apps, the Pod is the basic building block of deployment, scaling operations, and upgrades.
+
+Cloud native applications consist of sets of cloud native microservices. The configuration of these microservices is not changed after deployment (note, this does allow for changes in application data such as "add a new user"). New features for a cloud native application and the underlying Pods are rolled out as new artefacts and new configuration. Cloud native systems are configured declaratively. This means that the system configuration declares “what” a loosely coupled system should look like, not ”how” the system should be created. The “how” of the application is determined by the tooling (e.g. the orchestrator, operators, and CRDs).
+
+<a name="1.4"></a>
+## 1.4 Cloud Native Network Functions
```

Comment:
A CNF is a cloud native concept and should focus on cloud native principles as the basis.  Supporting additional requirements in an application, system or platform is a super set after first meeting the cloud native principles.   This section should focus on that and let other documents extend this base CNCF definition

Reply:
One thing to maybe keep in mind is that while the microservices configuration is not changed throughout their lifecycle, a telecom application (built up of many microservices) might need to expose configuration on  a higher abstraction level. For example you might want to change the general behavior of a CNF in the network (represented via a configuration management object exposed vie a nort-bound interface to a management system). This setting does not directly translate to "microservice configuration", still needs to be accessed, stored, managed, and still determimes the behavior of the CNF.

Reply:
Yes, I can see higher-level abstractions of configuration and management as something that could be desired in larger applications.  That is possible while following cloud native principles (including composability in this case).

If I'm understanding correctly, then it seems like the behaviour can be managed by the external management system directly with the CNFs cloud native API or through a "configuration management" service which communicates with the CNFs and provides a higher level of abstraction for the management system.  That configuration management service should itself be cloud native of course for the entire "larger application" to be cloud native.

The CNF modifiable run-time behaviour should be adjusted via APIs.  State for restarts, scaling, etc, should be stored outside of the CNF (eg. adjust max sessions to 10k).

Reply:
In my view, and I tend to agree with the "immutable infra principles", a CNF deployment state (ie interfaces, mem, libraries, device drivers, version, etc.) cannot be modified at runtime. How it behaves when a packet enters into it could be totally reprogrammable. For example, I could run a CNF Data Plane with P4Runtime which makes its forwarding pipeline totally modifiable but which does not change its nature to the underlying infra.

Reply:
Added a ticket to followup on this https://github.com/cncf/telecom-user-group/issues/34